### PR TITLE
Remove highlight, disabled, and iron curtain palettes

### PR DESCRIPTION
**Pull request by pchote**
This PR takes the first step towards replacing palette-based effects with RGBA-compatible techniques.

The first three commits do some basic reorganization and expands the core rendering plumbing to support per-sprite alpha and a new tint replacement mode. The last two commits (exactly) recreate the current target flash and WithColoredOverlay effects using tint/alpha and removes the now unnecessary palette definitions.

### CODE REVIEW

**Comment by pchote** [comment]
Rebased and adjusted update rule to account for the prep branch.

File: OpenRA.Mods.Common/Traits/Modifiers/WithColoredOverlay.cs [hunk 1]
```diff
@@ -45,10 +51,11 @@ IEnumerable<IRenderable> ModifiedRender(Actor self, WorldRenderer wr, IEnumerabl
 			{
 				yield return a;
 
-				if (!a.IsDecoration)
-					yield return a.WithPalette(wr.Palette(Info.Palette))
-						.WithZOffset(a.ZOffset + 1)
-						.AsDecoration();
+				if (!a.IsDecoration && a is IModifyableRenderable)
```
Comment (teinarss) [inline OpenRA.Mods.Common/Traits/Modifiers/WithColoredOverlay.cs]: not sure if we should start using `if (!a.IsDecoration && a is IModifyableRenderable ma)`

Comment (pchote) [reply OpenRA.Mods.Common/Traits/Modifiers/WithColoredOverlay.cs]: We should yeah, will fix.

**Comment by pchote** [comment]
Updated.

File: OpenRA.Mods.Common/Traits/Modifiers/WithColoredOverlay.cs [hunk 2]
```diff
@@ -45,10 +51,8 @@ IEnumerable<IRenderable> ModifiedRender(Actor self, WorldRenderer wr, IEnumerabl
 			{
 				yield return a;
 
-				if (!a.IsDecoration)
-					yield return a.WithPalette(wr.Palette(Info.Palette))
-						.WithZOffset(a.ZOffset + 1)
-						.AsDecoration();
+				if (!a.IsDecoration && a is IModifyableRenderable ma)
```
Comment (phrohdoh) [inline OpenRA.Mods.Common/Traits/Modifiers/WithColoredOverlay.cs:54]: Any particular reason to name this `ma` instead of `mr`?

Comment (pchote) [reply OpenRA.Mods.Common/Traits/Modifiers/WithColoredOverlay.cs:54]: we're looping over `for (var a in r)`, so `mr` would be misleading.

**Comment by teinarss** [comment]
needs a rebase.

**Comment by pchote** [comment]
Is there anything more I can do to help get this (and #18728) merged? They have been ready for months, and are major blockers for pretty much everything that I want to work on, which includes several long-wanted modding features.

**Review by teinarss — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/c039fb546895ba0b2a68463ca6dd6807a65daef4...74b167fc02bd7d871755a75359822eb4d1030ac9)

**Pull request merged**